pkg/api/types.go

```diff
@@ -193,6 +193,126 @@ type VolumeSource struct {
 	Secret *SecretVolumeSource `json:"secret"`
 	// NFS represents an NFS mount on the host that shares a pod's lifetime
 	NFS *NFSVolumeSource `json:"nfs"`
+	// PersistentVolumeClaimVolumeSource represents a reference to a PersistentVolumeClaim in the same namespace
+	PersistentVolumeClaimVolumeSource *PersistentVolumeClaimVolumeSource `json:"persistentVolumeClaim,omitempty"`
+}
+
+// Similar to VolumeSource but meant for the administrator who creates PVs.
+// Exactly one of its members must be set.
```

Comment:
I don't yet fully understand the thinking behind having only administrators create PV's, but I'm assuming that in future we're highly likely to allow normal K8S users to create PV's?  If this is true, do we want to pollute our API with a hard distinction between users and storage administrators?